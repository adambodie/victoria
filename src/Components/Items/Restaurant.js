import React from 'react'

const Restaurant = ({title, image, address, paragraph}) => (
	<div className='restaurant'>
		<h2 className='restaurantTitle'>{title}</h2>
		<div className='restaurantImage'>
			<img src={`..//assets/${image}.jpg`} alt={title} className='img-fluid'/>
			<div className='address'>
				{address.map((x, index) => <p key={index}>{x}</p>)}
			</div>
		</div>
		<div className='paragraph'>
			{paragraph.map((x, index) =><p key={index}>{x}</p>)}
		</div>
	</div>
)

export default Restaurant
