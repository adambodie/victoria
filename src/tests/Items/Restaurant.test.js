import React from 'react';
import renderer from 'react-test-renderer';
import Restaurant from '../../Components/Items/Restaurant';

it('renders a restaurant component', () => {
	const tree = renderer.create(
					<Restaurant 
						title="Restaurant Name" 
						image="restaurant image"
						address={["123 Fake Street", "Springfield, USA"]}
						paragraph={["One paragraph", "Two paragraph", "Three paragraph"]}
						/>
						).toJSON();
	expect(tree).toMatchSnapshot();
});
