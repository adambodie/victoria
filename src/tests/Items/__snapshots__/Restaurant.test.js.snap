// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a restaurant component 1`] = `
<div
  className="restaurant"
>
  <h2
    className="restaurantTitle"
  >
    Restaurant Name
  </h2>
  <div
    className="restaurantImage"
  >
    <img
      alt="Restaurant Name"
      className="img-fluid"
      src="..//assets/restaurant image.jpg"
    />
    <div
      className="address"
    >
      <p>
        123 Fake Street
      </p>
      <p>
        Springfield, USA
      </p>
    </div>
  </div>
  <div
    className="paragraph"
  >
    <p>
      One paragraph
    </p>
    <p>
      Two paragraph
    </p>
    <p>
      Three paragraph
    </p>
  </div>
</div>
`;
